compiler/rustc_mir_build/src/build/matches/mod.rs

```diff
@@ -690,6 +690,14 @@ impl<'a, 'tcx> Builder<'a, 'tcx> {
         visibility_scope
     }
 
+    crate fn check_var_scope(&self, var: HirId, span: Span) -> bool {
+        if !self.region_scope_tree.var_has_scope(var.local_id) {
+            self.tcx.sess.delay_span_bug(span, &format!("scope doesn't exist for {}", var));
+            return true;
+        }
+        false
+    }
+
     crate fn storage_live_binding(
```

Comment:
I think I would prefer to make `var_scope` return an `Option` and skip the drop scheduling within `storage_live_binding` if there is no scope.

This may cause further fallout down the road, but if we can do it I think it's preferrable over just skipping the assignment and storage live entirely

Reply:
Yeah, that makes more sense made the changes you suggested.